### DISCUSSION TITLE
setting `compactor_blocks_retention_period` did not reduce memory of gcs bucket

### DISCUSSION THREAD

**Discussion by SharifiFaranak** [grafana/mimir#5570] [Help and support] [answered]
#### Describe the bug
After setting `compactor_blocks_retention_period` to `395d` on our Mimir config, we did not see a drop in memory of our GCS bucket. 
![Screen Shot 2023-07-21 at 2 30 58 PM](https://github.com/grafana/mimir/assets/35040509/39e7eeed-d66f-4d70-9a9d-64c4a99a11b8)
![Screen Shot 2023-07-21 at 2 33 17 PM](https://github.com/grafana/mimir/assets/35040509/fc67fd27-aaae-408c-b5e7-610465b047f9)

#### To Reproduce
1. Run Mimir with GCS bucket as backend storage.
2. set `compactor_blocks_retention_period` to `395d`, or a shorter period of time.
3. observe that the size of the storage bucket doesn't decrease.

#### Expected behavior

Expect that there is a drop in the amount of memory that bucket is using. 

#### Environment

- Infrastructure: kubernetes
- Deployment tool: helm, argo

#### Additional Context

compactor logs:
```
level=info ts=2023-04-20T18:10:16.038473946Z caller=blocks_cleaner.go:467 component=cleaner user=score-prom msg="applied retention: marking block for deletion" block=01F3M398Z3GFXFG77HVGTHPVE9 maxTime=1618790400000
level=info ts=2023-04-20T18:10:16.184584177Z caller=block.go:203 component=cleaner user=score-prom msg="block has been marked for deletion" block=01F3M398Z3GFXFG77HVGTHPVE9
level=info ts=2023-04-20T18:10:16.800864408Z caller=blocks_cleaner.go:317 component=cleaner user=one-platform-q9 msg="completed blocks cleanup and maintenance" duration=1m25.023589354s
level=info ts=2023-04-20T18:10:41.28292092Z caller=blocks_cleaner.go:317 component=cleaner user=score-prom msg="completed blocks cleanup and maintenance" duration=1m37.149873862s
level=info ts=2023-04-20T18:10:41.282989169Z caller=blocks_cleaner.go:155 component=cleaner msg="successfully completed blocks cleanup and maintenance"
```

**Comment by 56quarters** [comment]
What makes you think the compactor isn't deleting things? It's marked something for deletion according to the logs you provided. After the `deletion_delay` has elapsed ([docs](https://grafana.com/docs/mimir/latest/references/configuration-parameters/#compactor)), the block will be removed.

**Reply by SharifiFaranak** [reply]
> What makes you think the compactor isn't deleting things?

In the event that the compactor did delete the blocks, I would expect the amount of memory in the gcs bucket to drop. I would expect to see some dip in the total bytes here:
![Screen Shot 2023-07-24 at 7 26 19 AM](https://github.com/grafana/mimir/assets/35040509/33f72f2b-1981-4e52-8f89-fffe948e0af7)

**Reply by pstibrany** [reply, marked as answer]
If you can access "tenants" and "tenant-blocks" endpoints on store-gateway (https://grafana.com/docs/mimir/latest/references/http-api/#store-gateway-tenants), can you check what are the oldest blocks on the storage and if they are older than 395 days?

**Reply by SharifiFaranak** [reply]
thank you @pstibrany, I was able to check that endpoint, and it seems that the oldest blocks there are not older than 365 days! I guess then my next question would be how to check that those blocks were deleted from the GCS bucket, if that is possible? 

The reason we configured the compactor to remove old metrics was to save on costs on the bucket.

**Reply by pstibrany** [reply]
> I guess then my next question would be how to check that those blocks were deleted from the GCS bucket, if that is possible?

If I understand correctly, you want to see if compactor would delete block that was older than 365 days? You can try to upload such a block, and let compactor discover it and then delete.

(If you have troubles finding such block, I've created one tiny block with max time from June 20, last year: https://gist.github.com/pstibrany/07daf58405e0e1d549fc8932830db716, you need to click "View raw" to download the zip file with the block, and then upload it to your bucket.)

**Reply by SharifiFaranak** [reply]
> If I understand correctly, you want to see if compactor would delete block that was older than 365 days?

Thank you for your help @pstibrany, I do think the compactor is deleting the correct blocks. 
I was just unsure why the total bytes of memory in the GCS bucket never dropped, after the old blocks were deleted. But I think you have answered my questions/I am able to confirm that this configuration is doing what it is supposed to, thank you so much.
